### DISCUSSION TITLE
Time Series Data

### DISCUSSION THREAD

**Discussion by TheAwesomeAndy** [BindsNET/bindsnet#512] [Q&A]
Hello,

Thank you for putting together this great project.  Do you have any examples, resources, or ideas for someone working with time series data classification rather than image data?

Thanks

**Comment by Hananel-Hazan** [comment]
Hi @TheAwesomeAndy 

Can you be more specific? what kind of data are you working on?
In the BindsNET there are couple of examples of games that can be use for time driven data.

**Comment by TheAwesomeAndy** [comment]
Thank you for your response!

I have physiological data including EEG, ECG, etc...  that are structured as sensor values captured over time.  The motivation being that SNNs have been mentioned as naturally being a good fit for time-series data.

Thanks!

**Reply by Hananel-Hazan** [reply]
Sound like a good dataset to use with Spiking network. For that you need to convert the analog EEG to a spike train, I suggest using Poisson encoder or something similar. Then using the reservoir framework with BindsNET to build a classifier. 
There is some example of reservoir at the MNIST example part that can help you started.

**Reply by TheAwesomeAndy** [reply]
Thank you for your reply, it is very helpful.  One difficulty I was having is how to replace MNIST with other data sources (most likely since I am new).  Do you have any examples or resources posted that show how to encode non-image data / Time series data?  

Thank you again for your help

**Reply by Hananel-Hazan** [reply]
I'm sorry, BindsNET doesn't have such an example. But, if you look at the reservoir example code, you can see that in line 139 the datum variable contains the spike train of the MNIST image. The spike train is the image converted to series of spikes using the Poisson encoder in line 104.
From here, you have a couple of options:

1. You can feed the data directly to the datum variable and for that, you need to define the network running time to the max size of your data. 
2. You can use some transformation to encode the EEG data to a spike train and then feed that input to the datum variable. 

In any case, the trick is to extract as much information you can from the analog data and converted it to a meaningful spike train that later on will help you to classify.

**Reply by TheAwesomeAndy** [reply]
Thanks so much for this very helpful feedback.  I will certainly give this a shot.  Hope I can come back with some good news about time series data -- should be interesting

**Reply by TheAwesomeAndy** [reply]
Hello,
Thanks for your help, to follow up on this question (sorry if I am asking an obvious question), would I have to create a wrapper similar to "MNIST" in order to encode and feed my ECG data into the reservoir?  Is there a simpler way -- currently my values are sensor values contained in a csv file.

Thanks for any insight

**Reply by Hananel-Hazan** [reply]
Maybe not exactly like the one from MNIST, but you need to convert your data to a spike train.
